### ISSUE TITLE
Listener not called after delete, since 7.0.0

### ISSUE THREAD

**Issue by BoD** [realm/realm-java#7188]
#### Goal
I have a query on all objects of a specific entity, and a listener on it. 

I'm deleting and inserting (in the same transaction) one object, on 6.1.0 my listener gets called, but not on 7.0.0. If I don't delete (just update), it works.

#### Actual Results
Listener not called

#### Steps & Code to Reproduce

Listener:
```
realm.where(User::class.java).findAll().addChangeListener(listener)
```

Delete/update:
```
    fun insertOrUpdateUser(user: User) = realm.executeTransaction {
        realm.delete(User::class.java) // <- removing this line works on 7.0.0 but why?
        realm.copyToRealmOrUpdate(user)
    }
```

My listener is kept in a reference (so kept active). On 7.0.0, it is not called when calling `insertOrUpdateUser`, whereas it is on 6.1.0.  When removing the delete, it is called for both versions. 

#### Version of Realm and tooling

Realm version(s): 6.1.0: works / 7.0.0

Realm Sync feature enabled: No

Android Studio version: 4.1.0

Android Build Tools version: 4.1.0

Gradle version: 6.7

Which Android version and device(s): reproduced on several devices/versions

**Comment by edualonso** [comment]
Hi @BoD. We believe this issue is caused by the way change listener notifications are triggered. If the notification token on a particular object is removed and replaced by another object (even in the same transaction, which is your case here), then the notification won't fire for the new object. I cannot give you an estimate on when it will be fixed, but I can see it's planned for Q4.

**Comment by BoD** [comment]
Thanks a lot for your reply.

This is troubling - on this particular case removing the deletion is acceptable (we have only one user object) but I'm afraid this will cause some unforeseen regressions in other places.

**Comment by edualonso** [comment]
Hi @BoD. If you really need to be notified about the two individual operations, you could split the transaction in two, one for the delete part and another for the insert part. I know it's far from optimal but at least it wouldn't break things elsewhere and it would fix your business logic problem until we have a fix for it.

**Comment by rorbech** [comment]
This seems to be a duplicate of #7098 which should be fixed in 10.3.0.
